service/src/test/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceTest.java

```diff
@@ -322,57 +334,142 @@ private BaseResource getResource() {
         List<PolicySet> onePolicySet = createOnePolicySet();
         List<PolicySet> twoPolicySets = createTwoPolicySets();
-        return new Object[][] {
-            { onePolicySet, null, onePolicySet },
-            { onePolicySet, Collections.emptyList(), onePolicySet },
-            { onePolicySet, Arrays.asList(onePolicySet.get(0).getName()), onePolicySet },
-            { twoPolicySets, Arrays.asList(twoPolicySets.get(0).getName()), twoPolicySets.subList(0, 1) },
-            { twoPolicySets, Arrays.asList(twoPolicySets.get(1).getName()), twoPolicySets.subList(1, 2) },
-            { twoPolicySets,
-                Arrays.asList(twoPolicySets.get(0).getName(), twoPolicySets.get(1).getName()), twoPolicySets }
+        return new Object[][] { filterNullPolicySetsListByNull(),
+            filterEmptyPolicySetsListByNull(),
+            filterOnePolicySetByNull(onePolicySet),
+            filterOnePolicySetByEmptyList(onePolicySet),
+            filterOnePolicySetByItself(onePolicySet),
+            filterTwoPolicySetsByFirstSet(twoPolicySets),
+            filterTwoPolicySetsBySecondPolicySet(twoPolicySets),
+            filterTwoPolicySetsByItself(twoPolicySets)
         };
     }
 
+    private Object[] filterTwoPolicySetsByItself(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets,
+                Arrays.asList(twoPolicySets.get(0).getName(), twoPolicySets.get(1).getName()), twoPolicySets };
+    }
+
+    private Object[] filterTwoPolicySetsBySecondPolicySet(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, Arrays.asList(twoPolicySets.get(1).getName()), twoPolicySets.subList(1, 2) };
+    }
+
+    private Object[] filterTwoPolicySetsByFirstSet(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, Arrays.asList(twoPolicySets.get(0).getName()), twoPolicySets.subList(0, 1) };
+    }
+
+    private Object[] filterOnePolicySetByItself(List<PolicySet> onePolicySet) {
+        return new Object[] { onePolicySet, Arrays.asList(onePolicySet.get(0).getName()), onePolicySet };
+    }
+
+    private Object[] filterOnePolicySetByEmptyList(List<PolicySet> onePolicySet) {
+        return new Object[] { onePolicySet, Collections.emptyList(), onePolicySet };
+    }
+
+    private Object[] filterOnePolicySetByNull(List<PolicySet> onePolicySet) {
+        return new Object[] { onePolicySet, null, onePolicySet };
+    }
+
+    private Object[] filterEmptyPolicySetsListByNull() {
+        return new Object[] { Collections.emptyList(), null, Collections.emptyList() };
+    }
+
+    private Object[] filterNullPolicySetsListByNull() {
+        return new Object[] { null, null, null };
+    }
+
     @DataProvider(name = "multiplePolicySetsRequestDataProvider")
     private Object[][] multiplePolicySetsRequestDataProvider()
             throws JsonParseException, JsonMappingException, IOException {
         List<PolicySet> onePolicySet = createOnePolicySet();
```

Comment:
- rename onePolicySet to denyPolicySet
- rename twoPolicySets to notApplicablePlusDenyPolicySet